pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java

```diff
@@ -270,8 +270,9 @@ public void start()
     // Initialize FunctionRegistry before starting the broker request handler
     FunctionRegistry.init();
     boolean caseInsensitive =
-        _brokerConf.getProperty(Helix.ENABLE_CASE_INSENSITIVE_KEY, false) || _brokerConf.getProperty(
-            Helix.DEPRECATED_ENABLE_CASE_INSENSITIVE_KEY, false);
+        _brokerConf.getProperty(Helix.ENABLE_CASE_INSENSITIVE_KEY, Helix.DEFAULT_ENABLE_CASE_INSENSITIVE)
```

Comment:
Not introduced in this PR, but the correct way to handle it should be:
1. Check if the key is explicitly configured
2. If so, use the value
3. If not, fall back to the deprecated key

Currently, if the official key is configured as `true`, but the deprecated key is configured as `false`, the overall result is `false` which is unexpected